### DISCUSSION TITLE
G29 Out of bounds

### DISCUSSION THREAD

**Discussion by SilberBur** [Jyers/Marlin#787] [Q&A] [answered]
Hello everyone

I've setup my Ender V2 with the 4.2.2 board with the pre compiled software. (Same issue no matter the version 5x5/3x3 HS/no HS)

However I had trouble to get a clean print with the UBL features.
I checked with Pronterface and get the following answer for a G29 L0/L1 command after creating and saving a mesh

SEND:G29 L0
? (L,R,F,B) out of bounds.

I've also noticed that sending the G29 T  and G29 A commands after homing (G28) start the probing precedure (not sure if that is correct?)

I've found the following site but I'm no where near understanding what is going on there.
https://github.com/MarlinFirmware/Marlin/issues/10538

Any help is welcome :D

**Comment by Tuncay-Ayhan** [comment]
> Hello everyone
> 
> I've setup my Ender V2 with the 4.2.2 board with the pre compiled software. (Same issue no matter the version 5x5/3x3 HS/no HS)
> 
> However I had trouble to get a clean print with the UBL features.
> I checked with Pronterface and get the following answer for a G29 L0/L1 command after creating and saving a mesh
> 
> SEND:G29 L0
> ? (L,R,F,B) out of bounds.
> 
> I've also noticed that sending the G29 T and G29 A commands after homing (G28) start the probing precedure (not sure if that is correct?)
> 
> I've found the following site but I'm no where near understanding what is going on there.
> [MarlinFirmware#10538](https://github.com/MarlinFirmware/Marlin/issues/10538)
> 
> Any help is welcome :D


<br>

<br>

Are you following a mix of guides? 

If you are using the firmware/display. You mesh is then saved in slot 0. That means you have to load slot 0 also. 


### If you want to load slot 0 

```
G29 L0
```

### if you want to load slot 1

```
G29 L1
```


#### I use UBL 10x10 HS
## This is my Cura startcode.



```
; Ender 3 Custom Start G-code
G92 E0 ; Reset Extruder
M190 S{material_bed_temperature_layer_0} ; preheat the bed
G28 ; autohome
G29 L0 ; load mesh from first slot
G29 J5 ; make autotilt for 5*5 grid
G29 F10 ; set Z fade to 10mm
G29 A ; enable UBL
M104 S{material_print_temperature_layer_0} ; set extruder temp
M105
M109 S{material_print_temperature_layer_0} ; wait for extruder temp

G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X0.1 Y20 Z0.3 F5000.0 ; Move to start position
G1 X0.1 Y200.0 Z0.3 F1500.0 E15 ; Draw the first line
G1 X0.4 Y200.0 Z0.3 F5000.0 ; Move to side a little
G1 X0.4 Y20 Z0.3 F1500.0 E30 ; Draw the second line
G92 E0 ; Reset Extruder
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X5 Y20 Z0.3 F5000.0 ; Move over to prevent blob squish
```

**Comment by Jyers** [comment, marked as answer]
> Same issue no matter the version 5x5/3x3 HS/no HS
> G29 T and G29 A commands after homing (G28) start the probing precedure

You're not using a UBL config, you're using bilinear. If you want UBL you need to use a firmware file specifically labeled with UBL
